src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -690,19 +690,26 @@ public ItemStack getIslandCrystal(int amount) {
         ItemStack itemStack = ItemStackUtils.makeItem(IridiumSkyblock.getInstance().getConfiguration().islandCrystal, Collections.singletonList(
                 new Placeholder("amount", String.valueOf(amount))
         ));
-        NBTItem nbtItem = new NBTItem(itemStack);
-        NBTCompound nbtCompound = nbtItem.getOrCreateCompound("iridiumskyblock");
-        nbtCompound.setInteger("islandCrystals", amount);
-        return nbtItem.getItem();
+
+        NBT.modify(itemStack, readWriteItemNBT -> {
+            readWriteItemNBT.getOrCreateCompound("iridiumskyblock").setInteger("islandCrystals", amount);
+        });
+
+        return itemStack;
     }
 
     public int getIslandCrystals(ItemStack itemStack) {
         if (itemStack == null || itemStack.getType() == Material.AIR) return 0;
-        NBTCompound nbtCompound = new NBTItem(itemStack).getOrCreateCompound("iridiumskyblock");
-        if (nbtCompound.hasKey("islandCrystals")) {
-            return nbtCompound.getInteger("islandCrystals");
-        }
-        return 0;
+
+        int amount = 0;
+
+        if (NBT.get(itemStack, readableItemNBT -> {
+            return readableItemNBT.getCompound("iridiumskyblock").hasTag("islandCrystals");
```

Comment:
can`t you combine both? would make more sense

Reply:
yeah probably but i couldnt figure it out without the IDE complaining

Reply:
feel free to post a suggestion

Reply:
return NBT.get(itemStack, readableItemNBT -> {
           if(readableItemNBT.getCompound("iridiumskyblock").hasTag("islandCrystals"))
            return readableItemNBT.getCompound("iridiumskyblock").getInteger("islandCrystals");
            return 0;
        });